torch/csrc/dynamo/compiled_autograd.h

```diff
@@ -236,6 +252,41 @@ class CompiledNodeArgs {
     collect(t.first);
     collect(t.second);
   }
+  template <typename V>
+  void collect(const ska::flat_hash_map<std::string, V>& m) {
+    collect_size(m.size());
+
+    std::vector<std::string> keys;
+    keys.reserve(m.size());
+    std::transform(
+        m.begin(), m.end(), std::back_inserter(keys), [](const auto& entry) {
+          return entry.first;
+        });
+    // stable sort to ensure deterministic collect order
+    std::stable_sort(keys.begin(), keys.end());
```

Comment:
Not all types will be sortable.  Which variable is triggering this?  Is it even needed?

Reply:
https://github.com/pytorch/pytorch/blob/368f242e377108a29db9ae821126a9df4b6e40a0/torch/csrc/autograd/custom_function.h#L111-L113

This one is pretty important, it contains all non-variables, fbgemm example: https://fburl.com/code/1bpdcp0s.

Reply:
This is only implemented for string keys, so it should always be sortable

Reply:
Let's add a static_assert (or just change the template) to only support string keys to be safe.